code/game/turfs/simulated/floor/light_floor.dm

```diff
@@ -80,3 +80,16 @@
 /turf/simulated/floor/light/extinguish_light()
 	toggle_light(FALSE)
 	visible_message("<span class='danger'>[src] flickers and falls dark.</span>")
+
+/turf/simulated/floor/light/clean(floor_only)
+	for(var/obj/effect/decal/cleanable/C in src)
+		var/obj/effect/decal/cleanable/blood/B = C
+		if(istype(B) && B.off_floor)
+			floor_only = FALSE
+		else
+			qdel(C)
+	if(floor_only)
+		clean_blood()
```

Comment:
I would honestly rather a 
```dm
var/colour_cache = color
..()
color = colour_cache
```
here, as that would cut down on the duped code, and make maintaining this easier.

Reply:
Got it, also the fact that you used both spellings of color at the same time is quite something.

Reply:
Mix of BYOND builtins and my native spelling